xapi/xapi.go

```diff
@@ -0,0 +1,166 @@
+package xapi
+
+import (
+	"io"
+	"io/ioutil"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/types"
+	"github.com/pingcap/tidb/xapi/tablecodec"
+	"github.com/pingcap/tidb/xapi/tipb"
+)
+
+// SelectResult is used to get response rows from SelectRequest.
+type SelectResult struct {
+	columns []*tipb.ColumnInfo
+	resp    kv.Response
+}
+
+// Next returns the next row.
+func (r *SelectResult) Next() (subResult *SubResult, err error) {
+	var reader io.ReadCloser
+	reader, err = r.resp.Next()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if reader == nil {
+		return nil, nil
+	}
+	subResult = &SubResult{
+		columns: r.columns,
+		reader:  reader,
+	}
+	return
+}
+
+// Close closes SelectResult.
+func (r *SelectResult) Close() error {
+	return r.resp.Close()
+}
+
+// SubResult represents a subset of select result.
+type SubResult struct {
+	columns []*tipb.ColumnInfo
```

Comment:
For "select a+b from t;", there is no physical column. How about use a struct like ResultField?